File: 再帰関数/数字の分割.py
def recursiveDigitsAdded(digits):
    return digits if digits < 10 else recursiveDigitsAddedHelper(digits, 0)
def recursiveDigitsAddedHelper(digits, res):
    if digits < 10:
        return res 
    sum_num = 0
    while digits > 0:
        sum_num += digits % 10
        digits //= 10  
    return recursiveDigitsAddedHelper(sum_num, res + sum_num)

File: PHP/CS中級/数字の分割.php
<?php

function recursiveDigitsAdded(int $digits): int{
    return $digits < 10 ? $digits : recursiveDigitsAddedHelper($digits, 0, 0);
}

function recursiveDigitsAddedHelper(int $digits, int $current,int $ans): int{
    if ($digits < 10) return $ans;

    while ($digits > 0) {
        $current += $digits % 10;
        $digits = floor($digits / 10);
        echo $current . PHP_EOL;
    }

    return recursiveDigitsAddedHelper($current, 0, $ans += $current);
}